### PR TITLE
#244 JUnit* rules are not compatible with JUnit5

### PR DESCRIPTION
**Pull request by ychulovskyy**

### CODE REVIEW

File: src/main/groovy/org/codenarc/rule/junit/JUnitPublicNonTestMethodRule.groovy [hunk 1]
```diff
@@ -50,6 +50,13 @@ class JUnitPublicNonTestMethodRule extends AbstractAstVisitorRule {
     int priority = 2
     Class astVisitorClass = JUnitPublicNonTestMethodAstVisitor
     String applyToClassNames = DEFAULT_TEST_CLASS_NAMES
```
Comment (chrismair) [inline src/main/groovy/org/codenarc/rule/junit/JUnitPublicNonTestMethodRule.groovy]: The fact that *ignoreMethodsWithAnnotations* is a property means it is publicly accessible. It is confusing to have a setter with the same name but different type. Did you intend this to be *overloaded*? It seems like either *ignoreMethodsWithAnnotations* should be a private field or else the property should have a different name.

File: src/site/apt/codenarc-rules-junit.apt [hunk 2]
```diff
@@ -208,9 +208,11 @@ JUnit Rules  ("<rulesets/junit.xml>")
...
 
-      * Methods annotated with <<<@Before>>> and <<<@After>>>
+      * Methods annotated with <<<@Before>>>, <<<@BeforeAll>>>, <<<@BeforeClass>>> and <<<@BeforeEach>>>
 
-      * Methods annotated with <<<@BeforeClass>>> and <<<@AfterClass>>>
+      * Methods annotated with <<<@After>>>, <<<@AfterAll>>>, <<<@AfterClass>>> and <<<@AfterEach>>>
+
+      * Methods annotated with <<<@Disabled>>> and <<<@Ignore>>>
 
       * Methods annotated with <<<@Override>>>
```
Comment (chrismair) [inline src/site/apt/codenarc-rules-junit.apt]: Can you add a table row for the rule configuration property (*ignoreMethodsWithAnnotations*).

File: src/main/groovy/org/codenarc/rule/junit/JUnitPublicNonTestMethodRule.groovy (hunk 1 above)
Comment (ychulovskyy) [reply src/main/groovy/org/codenarc/rule/junit/JUnitPublicNonTestMethodRule.groovy]: Good catch! Fixed

File: src/site/apt/codenarc-rules-junit.apt (hunk 2 above)
Comment (ychulovskyy) [reply src/site/apt/codenarc-rules-junit.apt]: Fixed

**Comment by ychulovskyy** [comment]
@chrismair Happy New Year!

**Comment by chrismair** [comment]
Thanks for the contribution!

**Pull request merged**